compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1806,6 +1806,12 @@ rustc_queries! {
         desc { "looking up lifetime defaults for generic parameter `{}`", tcx.def_path_str(def_id) }
         separate_provide_extern
     }
+
+    query get_default_impl_equivalent(def_id: DefId) -> Option<DefId> {
+        desc { |tcx| "looking up the unit variant or fn call with no arguments equivalence for the `Default::default()` of `{}`", tcx.def_path_str(def_id) }
```

Comment:
i'm having a really hard understanding what this means

Reply:
also i don't think this descr is grammatical. do you mean "... with no arguments equivalent to the `Default::default()`"?

Reply:
I have no idea whether this takes an impl def id or a struct def id or what.